process/throttle/antiflood/p2pBlackListProcessor.go

```diff
@@ -0,0 +1,108 @@
+package antiflood
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type p2pBlackListProcessor struct {
+	cacher                     storage.Cacher
+	blacklistHandler           process.BlackListHandler
+	thresholdNumReceivedFlood  uint32
+	thresholdSizeReceivedFlood uint64
+	numFloodingRounds          uint32
+}
+
+// NewP2pBlackListProcessor creates a new instance of p2pQuotaBlacklistProcessor able to determine
+// a flooding peer and mark it accordingly
+func NewP2pBlackListProcessor(
+	cacher storage.Cacher,
+	blacklistHandler process.BlackListHandler,
+	thresholdNumReceivedFlood uint32,
+	thresholdSizeReceivedFlood uint64,
+	numFloodingRounds uint32,
+) (*p2pBlackListProcessor, error) {
+
+	if check.IfNil(cacher) {
+		return nil, fmt.Errorf("%w, NewP2pBlackListProcessor", process.ErrNilCacher)
+	}
+	if check.IfNil(blacklistHandler) {
+		return nil, fmt.Errorf("%w, NewP2pBlackListProcessor", process.ErrNilBlackListHandler)
+	}
+	if thresholdNumReceivedFlood == 0 {
+		return nil, fmt.Errorf("%w, thresholdNumReceivedFlood == 0", process.ErrInvalidValue)
+	}
+	if thresholdSizeReceivedFlood == 0 {
+		return nil, fmt.Errorf("%w, thresholdSizeReceivedFlood == 0", process.ErrInvalidValue)
+	}
+	if numFloodingRounds == 0 {
+		return nil, fmt.Errorf("%w, numFloodingRounds == 0", process.ErrInvalidValue)
+	}
+
+	return &p2pBlackListProcessor{
+		cacher:                     cacher,
+		blacklistHandler:           blacklistHandler,
+		thresholdNumReceivedFlood:  thresholdNumReceivedFlood,
+		thresholdSizeReceivedFlood: thresholdSizeReceivedFlood,
+		numFloodingRounds:          numFloodingRounds,
+	}, nil
+}
+
+// ResetStatistics checks if an identifier reached its maximum flooding rounds. If it did, it will remove its
+// cached information and adds it to the black list handler
+func (pbp *p2pBlackListProcessor) ResetStatistics() {
+	keys := pbp.cacher.Keys()
+	for _, key := range keys {
+		obj, ok := pbp.cacher.Peek(key)
+		if !ok {
+			pbp.cacher.Remove(key)
```

Comment:
cacher.remove is used at every continue and if numFloodinRounds is high. should be refactored to be called only once.

Reply:
I know, could not find any proper way to do that. I also did not want to have
```GO
if .... {
   if ....{
   }
}
```